src/common/components/Embeds/OpenGraphImage.tsx

```diff
@@ -0,0 +1,35 @@
+import React, { useEffect, useState } from 'react';
+
+const OpenGraphImage = ({ url }: { url: string }) => {
+  const [metadata, setMetadata] = useState(null);
+
+  useEffect(() => {
+    const fetchMetadata = async () => {
+      const request = await fetch("https://api.modprotocol.org/api/cast-embeds-metadata/by-url", {
+        body: JSON.stringify([url]),
+        method: 'POST',
+        headers: {
+          'Content-Type': "application/json"
+        }
+      });
+      const metadata = await request.json();
```

Comment:
the return data in `metadata` has the following example format for the url "https://summerofprotocols.com/research/re-move"
```
{
    "https://summerofprotocols.com/research/re-move": {
        "image": {
            "url": "https://summerofprotocols.com/wp-content/uploads/2023/12/Re-Move-Header-jpg.webp",
            "height": 2048,
            "width": 2048
        },
        "description": "Imagine navigating memories stored on circular hard disks, much like flipping through room numbers in a building. Just as people rely on room numbers to find their way, a comparable process is used to locate specific memories on these disks by computers. In this visual narrative, to recover fuzzy",
        "title": "Re-Move",
        "publisher": "Summer of Protocols"
    }
}
```

fix the line `setMetadata(metadata);` accordingly